internal/model/todos.go

```diff
@@ -19,81 +19,119 @@ type TodoList struct {
 	Todos []*Todo `json:"todos"`
 }
 
-// GetTodosListByID returns slice of Todos based on the listID passed in
-func GetTodosListByID(listID string) (*TodoList, error) {
-	listStatement := `SELECT id, name FROM lists WHERE id = $1`
+// TodoListByID fetches a list and its todos
+func TodoListByID(id string) (*TodoList, error) {
 	list := &TodoList{}
 
-	err := db.QueryRow(listStatement, listID).Scan(&list.ID, &list.Name)
+	sql := "SELECT id, name FROM lists WHERE id = $1"
+	err := db.QueryRow(sql, id).Scan(&list.ID, &list.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	sqlStatement := `SELECT id, list_id, todo, done FROM todos WHERE list_id = $1 ORDER BY created_at ASC;`
+	list.Todos, err = TodosByListID(id)
 
-	rows, err := db.Query(sqlStatement, listID)
+	return list, nil
+}
+
+func (tl *TodoList) validate() error {
+	if tl.Name == "" {
+		return errors.New("name empty, must be set")
+	}
+	if len(tl.Todos) == 0 {
+		return errors.New("no todos, must have at least one")
+	}
+
+	return nil
+}
+
+// Save saves a new list to the database
+func (tl *TodoList) Save() error {
+	if err := tl.validate(); err != nil {
+		return err
+	}
+	sql := `INSERT INTO lists(name) VALUES($1) RETURNING id;`
+	err := db.QueryRow(sql, tl.Name).Scan(&tl.ID)
```

Comment:
So now we have a `Save` method *on* the `TodoList` struct. This keeps us from having to pass in a bunch of variables and makes the api easy as you'll see later.

`validate` is an un-exported method (can only be called within the db package), and is a nice place to validate our model before we can save it.